src/backend/core/api/serializers.py

```diff
@@ -234,8 +234,11 @@ def validate_file(self, file):
 
         # Try guessing a coherent extension from the mimetype
         if extension_mime_type != magic_mime_type:
-            if guessed_ext := mimetypes.guess_extension(magic_mime_type):
-                extension = guessed_ext[1:]
+            self.context["is_unsafe"] = True
+
+        guessed_ext = mimetypes.guess_extension(magic_mime_type)
+        if (extension is None or len(extension) > 5) and guessed_ext:
```

Comment:
Why you need to check `len(extension) > 5`? Is it because no extension should be longer? Maybe a quick comment can help understand here.

Reply:
yep. It seems that 5 characters is as long as it gets. Will add comment.